utils/src/main/scala/za/co/absa/enceladus/utils/broadcast/BroadcastUtils.scala

```diff
@@ -125,70 +104,68 @@ object BroadcastUtils {
 
     val lambda = getErrorLambda(mappingTable, outputColumns, mappings, numberOfArguments)
 
-    val errorMessageSchema = ScalaReflection.schemaFor[ErrorMessage]
-
-//    val errorUdf = UserDefinedFunction(lambda,
-//      errorMessageSchema.dataType,
-//      Some(mappingTable.value.keyTypes))
-//
-//    errorUdf
-
-    // todo fix Issue #1710
-    throw new NotImplementedError("This method is not correctly implemented, yet")
+    lambda
   }
 
-  private def getSingleMappingLambda(mappingTable: Broadcast[LocalMappingTable],
-                                     defaultValueMap: Map[String, Any], numberOfArguments: Int): AnyRef = {
-    val defaultValue = defaultValueMap.get(mappingTable.value.outputColumns.values.head).orNull
-    numberOfArguments match {
-      case 1 => (p1: Any) => mappingTable.value.getRowWithDefault(Seq(p1), defaultValue)
-      case 2 => (p1: Any, p2: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2), defaultValue)
-      case 3 => (p1: Any, p2: Any, p3: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2, p3), defaultValue)
-      case 4 => (p1: Any, p2: Any, p3: Any, p4: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4), defaultValue)
-      case 5 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5), defaultValue)
-      case 6 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6), defaultValue)
-      case 7 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7), defaultValue)
-      case 8 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8), defaultValue)
-      case 9 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any, p9: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8, p9), defaultValue)
-      case 10 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any, p9: Any, p10: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10), defaultValue)
-      case n => throw new IllegalArgumentException(s"Mapping UDFs with $n arguments are not supported. Should be between 1 and 10.")
-    }
-  }
-
-  private def getMultipleMappingLambda(mappingTable: Broadcast[LocalMappingTable],
-                                       defaultValues: Map[String, Any], numberOfArguments: Int): AnyRef = {
-    val defaultRow = if (defaultValues.isEmpty) {
+  private def getMappingLambda(mappingTable: Broadcast[LocalMappingTable],
```

Comment:
I would refactor this considerably:
```
private def getLambda(numberOfArguments: Int, fnc: Seq[Any] => Any): UserDefinedFunction = {
...
}

private def getLambdaFnc(, defaultValue: Any)
```

In `getMappingUdfForSingleOutput`:
```
   ...
    val defaultValue = defaultValueExprMap.get(mappingTable.value.outputColumns.values.head).orNull
    val fnc: Seq[Any] => Any  = (seq: Seq[Any] ) => mappingTable.value.getRowWithDefault(seq, defaultValue)
    getLambda(numberOfArguments, fnc)   
```
or alternatively:
```
   ...
    val defaultValue = defaultValueExprMap.get(mappingTable.value.outputColumns.values.head).orNull
    getLambda(numberOfArguments, mappingTable.value.getRowWithDefault(_, defaultValue))   
```

Similarly in `getMappingUdfForMultipleOutputs` and `getErrorUdf`

Reply:
Good idea, I will also need to add the valueType parameter